lib/extract/extractTransform.js

```diff
@@ -13,71 +14,172 @@ function transformToMatrix(props, transform) {
+    {
+        return [
+            1, 0, tx,
+            0, 1, ty || 0
+        ];
+    }
+
+scale
+    = "scale" wsp* "(" wsp* sx:number sy:commaWspNumber? wsp* ")"
+    {
+        return [
+            sx, 0,                     0,
+            0,  sy === null ? sx : sy, 0
+        ];
+    }
+
+rotate
+    = "rotate" wsp* "(" wsp* angle:number c:commaWspTwoNumbers? wsp* ")"
+    {
+        const cos = Math.cos(deg2rad * angle);
+        const sin = Math.sin(deg2rad * angle);
+        if (c !== null) {
+            const [x, y] = c;
+            return [
+                cos, -sin, cos * -x + -sin * -y + x,
+                sin,  cos, sin * -x +  cos * -y + y
+            ];
+        }
+        return [
+            cos, -sin, 0,
+            sin,  cos, 0
+        ];
+    }
+
+skewX
+    = "skewX" wsp* "(" wsp* angle:number wsp* ")"
+    {
+        return [
+            1, Math.tan(deg2rad * angle), 0,
+            0, 1,                         0
+        ];
+    }
+
+skewY
+    = "skewY" wsp* "(" wsp* angle:number wsp* ")"
+    {
+        return [
+            1,                         0, 0,
+            Math.tan(deg2rad * angle), 1, 0
+        ];
+    }
+
+number
+    = f:(sign? floatingPointConstant) { return parseFloat(f.join("")); }
+    / i:(sign? integerConstant) { return parseInt(i.join("")); }
+
+commaWspNumber
+    = commaWsp n:number { return n; }
+
+commaWspTwoNumbers
+    = commaWsp n1:number commaWsp n2:number { return [n1, n2]; }
+
+commaWsp
+    = (wsp+ comma? wsp*) / (comma wsp*)
+
+comma
+    = ","
+
+integerConstant
+    = ds:digitSequence { return ds.join(""); }
+
+floatingPointConstant
+    = fractionalConstant exponent?
+    / digitSequence exponent
 
+    fractionalConstant "fractionalConstant"
+    = d1:digitSequence? "." d2:digitSequence { return [d1 ? d1.join("") : null, ".", d2.join("")].join(""); }
+    / d:digitSequence "." { return d.join(""); }
+
+exponent
+    =  [eE] sign? digitSequence
+
+sign
+    = [+-]
+
+digitSequence
+    = digit+
+
+digit
+    = [0-9]
+
+wsp
+    = [\\u0020\\u0009\\u000D\\u000A]
+`);
 
 function appendTransform(transform) {
     if (transform) {
         if (typeof transform === 'string') {
-            const transformParsed = tp.parse(transform);
-            if (transformParsed.matrix) {
-                pooledMatrix.append(...transformParsed.matrix);
-            }
-            else {
-                let trans = props2transform(transformParsed);
-                if (typeof trans !== 'string') {
-                    transform = trans;
-                }
+            try {
+                const [a, c, e, b, d, f] = transformParser.parse(transform);
+                pooledMatrix.append(...[a, b, c, d, e, f]);
+            } catch (e) {
+                console.error(e);
             }
-        }
-        if (typeof transform !== 'string') {
+        } else {
             pooledMatrix
                 .appendTransform(
```

Comment:
This part, props2transform, transformToMatrix & universal2axis should probably be removed. They're not part of the svg spec and a source of confusion and bugs, as the order of transforms being applied isn't documented nor specified anywhere except the source code of the implementation, and is quite arbitrary. What do you think @magicismight is such a big breaking change possible? To align it with the evergreen browsers?